### PR TITLE
Suggestion: Added basic architecture for common tests with cmd arguments

### PR DESCRIPTION
**Pull request by ahcorde**
Signed-off-by: ahcorde <ahcorde@gmail.com>

# 🎉 New feature

## Summary

Use command line arguments to setup the physic engine library.

FYI @scpeters 

## Checklist
- [ ] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: test/common_test/basic_test.cc [hunk 1]
```diff
@@ -34,23 +34,30 @@ class EntityManagementFeaturesTest:
...
+    if (argc != 2)
+      FAIL() << "Please provide the path to an engine plugin.\n"
+             << "Usage COMMON_TEST_basic_test <physics engine path>\n";
+    libToTest = argv[1];
+  }
+
+  static std::string GetLibToTest()
+  {
+		return libToTest;
+	}
```
Comment (scpeters) [inline test/common_test/basic_test.cc:48]: nit: tabs to spaces on the 2 lines above

File: test/common_test/CMakeLists.txt [hunk 2]
```diff
@@ -7,12 +7,11 @@ set(tests
...
-  set(_env_vars)
-  add_test(NAME ${test_name}_${PHYSICS_ENGINE_NAME} COMMAND ${test_name})
-  list(APPEND _env_vars "LIB_TO_TEST=${CMAKE_BINARY_DIR}/lib/${CMAKE_SHARED_LIBRARY_PREFIX}${PROJECT_LIBRARY_TARGET_NAME}-${PHYSICS_ENGINE_NAME}-plugin${CMAKE_SHARED_LIBRARY_SUFFIX}")
-  set_tests_properties(${test_name}_${PHYSICS_ENGINE_NAME}
-    PROPERTIES
-      ENVIRONMENT "${_env_vars}")
+  add_test(NAME ${test_name}_${PHYSICS_ENGINE_NAME}
+    COMMAND
+      ${test_name}
+      ${CMAKE_BINARY_DIR}/lib/${CMAKE_SHARED_LIBRARY_PREFIX}${PROJECT_LIBRARY_TARGET_NAME}-${PHYSICS_ENGINE_NAME}-plugin${CMAKE_SHARED_LIBRARY_SUFFIX}
```
Comment (scpeters) [inline test/common_test/CMakeLists.txt:13]: nit: use a generator expression to get the library file path from a target name

**Review by scpeters — COMMENTED** [review]
Yes!!! This is exactly what I was hoping for!

File: test/common_test/CMakeLists.txt (hunk 2 above)
Comment (ahcorde) [reply test/common_test/CMakeLists.txt]: https://github.com/gazebosim/gz-physics/pull/359/commits/2d5ba810cab566e00a96f8292799f1c4dad0c365

**Review by scpeters — APPROVED** [review]

**Pull request merged**